runtime/src/qio/qio_formatted.c

```diff
@@ -360,6 +360,68 @@ qioerr _append_char(char* restrict * restrict buf, size_t* restrict buf_len, siz
 
   return 0;
 }
+qioerr read_entire_file(qio_channel_t* restrict ch, const char* restrict* restrict out, int64_t* restrict len_out, ssize_t maxlen)
+{
+    qioerr err,err_reading_file;
```

Comment:
indenting in this function is a bit weird. Please use two spaces per indent level and make sure each section in curly braces has a new indent level.

Reply:
Changes made as requested .